### ISSUE TITLE
[Sites] Add configuration to limit availability of pages in main site or parent site context

### ISSUE THREAD

**Issue by brusch** [pimcore/pimcore#7187]
Consider the following document tree structure: 
```text
- Home (Main Site)
  - Site 1
    - Page 4
    - Site 3
      - Page 1
      - Page 2
  - Site 2
```

If we take a look at `Page 4`, this page is currently available under the context of the main side as well as under `Site 1`. This is not always a wanted behavior and it even get worse if you have nested site structures (see `Page 1` and `Page 2`). 
Another challenge is that Pimcore isn't able to guess how the link (in which context) to e.g. `Page 4` should be generated, should it be `https://main.page/Site 1/Page 4` or should it be `https://site1.com/Page 4`. 

This is also related to #7139

My proposal is to add a new configuration option for sites, so specify whether the site should be available also in parent contexts or just in the context of the actual site. 

![image](https://user-images.githubusercontent.com/142037/94530560-8bfef500-023b-11eb-8707-3729746a717e.png)

**Important**: This needs to be implemented very carefully with performance in mind - this could have a huge performance impact if not done properly!

**Comment by dpfaffenbauer** [comment]
That is also SEO relevant with duplicate content. Do you have use-cases where you nest sites and also want to display sub-site content in the parent context?

**Comment by markus-moser** [comment]
Yes this behavior is a big potential problem and I'm glad that this get's solved.

@dpfaffenbauer I never had such a use case. In my opinion we should force the "Not accessible in parent contexts" or at least rename it to "Accessible in parent contexts" and set the default behavior to not accessible.

**Comment by brusch** [comment]
Fixed by #14706 and duplicates #14315
